API/controllers/postController.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const _ = require('lodash');
+
+require('../models/Posts');
+const Post = mongoose.model('Post');
+
+/**
+ * Methods:
+ *  post(/) => Creates new post
+ *  get(/:id) => Gets post by id
+ *  put(/:id) => Updates post with given id
+ *  delete(/:id) => Deletes post with given id
+ *  get(/:id/comments) => Gets all comments for post
+ *  post(/:id) => Adds a new comment to the post
+ */
+
+/**
+ * Create new post
+ */
+router.post('/', (req, res) => {
+  Post.create(req.body).then(post => {
+    res.json(200, post);
+  })
+  .catch(err => {
+    if (typeof err === 'object') {
+      res.json(500, err.message);
+    } else {
+      res.json(403, err);
+    }
+  })
+})
+
+/**
+ * Get post by id
+ */
+router.get('/:id', (req, res) => {
+  Post.get(req.params.id).then(post => {
+    res.json(200, post);
+  })
+  .catch(err => {
+    if (typeof err === 'object') {
+      res.json(500, err.msg);
+    } else {
+      res.json(403, err);
+    }
+  });
+});
+
+/**
+ * Update a post
+ */
+router.put('/:id', (req, res) => {
+  const id = req.params.id;
+  Post.updatePost(id, req.body).then(() => {
+    res.redirect(`/posts/${id}`);
```

Comment:
Is there a reason we're redirecting instead of just responding with the updated post?

Reply:
the update method doesn't actually return the new object, so we need to retrieve that somehow. Could wrap the whole thing in a findOneAndUpdate method instead (with `{new: true}` in the query params), might be better convention (although same number of db calls)

Reply:
I think I'd prefer that to a redirect. It's totally reasonable for an `update` operation to return the modified object, so we don't have to rename the method, and it'll save us a network request from the redirect, even if db calls are the same.